.github/workflows/ci.yml

```diff
@@ -487,7 +487,6 @@ jobs:
 
   test:
     name: Unit Testing
-    if: ${{ always() }}
```

Comment:
I think due to this line, unittests tries to get executed even if the build is failed, resulting occational hangs.